vernac/record.ml

```diff
@@ -703,15 +720,28 @@ let declare_structure { Record_decl.mie; primitive_proj; impls; globnames; globa
     let () = if is_coercion then ComCoercion.try_add_new_coercion build ~local:false ~poly ~nonuniform:false ~reversible:true in
     let struc = Structure.make (Global.env ()) rsp projections in
     let () = declare_structure_entry struc in
-    rsp
+    GlobRef.IndRef rsp
   in
-  List.mapi map records
+  List.mapi map records, []
 
-let implicits_of_context ctx =
-  List.map (fun name -> CAst.make (Some (name,true)))
-    (List.rev (Anonymous :: (List.map RelDecl.get_name ctx)))
-
-let build_class_constant ~univs ~rdata ~primitive_proj field implfs params paramimpls coers binder id proj_name =
+let get_class_params = function
+  | [{ Data.id; idbuild; rdata; is_coercion; coers; inhabitant_id }] ->
+     let coers = List.map (fun { Data.pf_subclass; pf_priority } -> if pf_subclass then Some { hint_priority = pf_priority; hint_pattern = None } else None) coers in
+     id, idbuild, rdata, is_coercion, coers, inhabitant_id
+  | _ ->
+     CErrors.user_err (str "Mutual definitional classes are not supported.")
+
+(* declare definitional class (typeclasses that are not record) *)
```

Comment:
Please let's profit and document the pre-conditions here.

Reply:
Done